lib/kafka_ex/protocol/common.ex

```diff
@@ -0,0 +1,16 @@
+defmodule KafkaEx.Protocol.Common do
+  @moduledoc """
+  A collection of common request generation and response parsing functions for the
+  Kafka wire protocol.
+  """
+
+  @doc """
+  Generate the wire representation for a list of topics.
+  """
+  def topic_data([]), do: ""
+
+  def topic_data([topic|topics]) do
```

Comment:
Oh, I see now that it wasn't tail-call recursive before.  As long as you're refactoring, it's probably a good idea to make that change.